src/Codeception/Test/Cept.php

```diff
@@ -0,0 +1,78 @@
+<?php
+namespace Codeception\Test;
+
+use Codeception\Exception\TestParseException;
+use Codeception\Lib\Parser;
+
+class Cept extends Test implements Interfaces\Plain, Interfaces\ScenarioDriven, Interfaces\Reported
+{
+    use Feature\ScenarioLoader;
+
+    /**
+     * @var Parser
+     */
+    protected $parser;
+
+    public function __construct($name, $file)
+    {
+        $this->setMetadata(new Metadata());
+        $this->getMetadata()->setName($name);
+        $this->getMetadata()->setFilename($file);
+        $this->createScenario();
+        $this->parser = new Parser($this->getScenario(), $this->getMetadata());
+    }
+
+    public function preload()
+    {
+        $this->getParser()->prepareToRun($this->getRawBody());
+    }
+
+    public function test()
+    {
+        $scenario = $this->getScenario();
+        $testFile = $this->getMetadata()->getFilename();
```

Comment:
I was thinking on making Metadata private... To avoid side effects if someone introduce such property in test of `Codeception\Test\TestCase` and change it.